stac_fastapi/elasticsearch/core.py

```diff
@@ -164,7 +165,15 @@ def get_queryset(self, **kwargs) -> Search:
                     )
                 )
 
-        return qs
+        matched = qs.count()
+        if self.extension_is_enabled('ContextExtension'):
```

Comment:
It would be good to move this out of the `get_queryset` method. In the world where we have those hooks for the extensions to modify the queryset, we would have to recalculate anyway. I would put this after the queryset has been returned.